examples/asr/asr_streaming/speech_to_text_streaming_infer.py

```diff
@@ -0,0 +1,333 @@
+        asr_model = nemo_asr.models.ASRModel.from_pretrained(model_name=args.asr_model)
+
+    logging.info(asr_model.encoder.streaming_cfg)
+
+    global autocast
+    if (
+        args.use_amp
+        and torch.cuda.is_available()
+        and hasattr(torch.cuda, 'amp')
+        and hasattr(torch.cuda.amp, 'autocast')
+    ):
+        logging.info("AMP enabled!\n")
+        autocast = torch.cuda.amp.autocast
+    else:
+
+        @contextlib.contextmanager
+        def autocast():
+            yield
+
+    # configure the decoding config
+    decoding_cfg = asr_model.cfg.decoding
+    with open_dict(decoding_cfg):
+        decoding_cfg.strategy = "greedy"
+        decoding_cfg.preserve_alignments = False
+        if hasattr(asr_model, 'joint'):  # if an RNNT model
+            decoding_cfg.greedy.max_symbols = 10
+            decoding_cfg.fused_batch_size = -1
+        asr_model.change_decoding_strategy(decoding_cfg)
+
+    asr_model = asr_model.to(args.device)
+    asr_model.eval()
+
+    # In streaming, offline normalization is not feasible as we don't have access to the whole audio at the beginning
+    # When online_normalization is enabled, the normalization of the input features (mel-spectrograms) are done per step
+    # It is suggested to train the streaming models without any normalization in the input features.
+    if args.online_normalization:
+        if asr_model.cfg.preprocessor.normalize not in ["per_feature", "all_feature"]:
+            logging.warning(
+                "online_normalization is enabled but the model has no normalization in the feature extration part, so it is ignored."
+            )
+            online_normalization = False
+        else:
+            online_normalization = True
+
+    else:
+        online_normalization = False
+
+    streaming_buffer = FramewiseStreamingAudioBuffer(model=asr_model, online_normalization=online_normalization)
+    if args.audio_file is not None:
+        # stream a single audio file
+        processed_signal, processed_signal_length, stream_id = streaming_buffer.append_audio_file(
+            args.audio_file, stream_id=-1
+        )
+        perform_streaming(
+            asr_model=asr_model, streaming_buffer=streaming_buffer, compare_vs_offline=args.compare_vs_offline
+        )
+    else:
+        # stream audio files in a manifest file in batched mode
+        samples = []
+        all_streaming_tran = []
+        all_offline_tran = []
+        all_refs_text = []
+
+        with open(args.manifest_file, 'r') as f:
+            for line in f:
+                item = json.loads(line)
+                samples.append(item)
+
+        logging.info(f"Loaded {len(samples)} from the manifest at {args.manifest_file}.")
+
+        start_time = time.time()
+        for sample_idx, sample in enumerate(samples):
+            processed_signal, processed_signal_length, stream_id = streaming_buffer.append_audio_file(
+                sample['audio_filepath'], stream_id=-1
```

Comment:
What does stream id -1 mean?

Reply:
It adds a new stream at the end of the streams and append this audio to the new stream. If a positive number is passed, it would append the audio at the end of that stream_id.

Reply:
We need to document this script a lot more in the branch cut for 1.11. For now its fine